src/ngraph/pattern/matcher.cpp

```diff
@@ -190,6 +190,16 @@ namespace ngraph
             add_node(graph_node);
             size_t watermark = m_matched_list.size() - 1;
 
+            // we will abort the match if any of graph_node or the pattern_node contains
+            // dynamic shape
+            if (m_strict_mode && graph_node->get_outputs().size() == 1)
+            {
+                if (pattern_node->is_dynamic() || graph_node->is_dynamic())
+                {
+                    return abort_match(watermark, false);
```

Comment:
shouldn't this be configurable instead of aborting on dynamic/partial shapes always?

Reply:
Actually after looking at this, I am not sure we should do this check here. In my other PR, it added property to `Matcher` as well. So a matcher should specify whether it can handle dynamic shape or not and the GraphRewrite pass should do that check against the `Function`.

@pruthviIntel Do you think this makes sense?

Reply:
To be more specific, the Matcher does not have problem with dynamic shapes (beside the strict mode, which checks shape). It's the *callback* may have problems with the dynamic shape.

Reply:
@louisfeng should we remove this check? or query `get_property()` and check if it has `REQUIRE_STATIC_SHAPE` &&  `pattern_node->is_dynamic()`  and abort?